File: RepoTransBench/target_projects/Python/C++/fengsp_sender/public_tests/public_test_sender_extended.cpp
#include <gtest/gtest.h>
#include "attachment.h"

class PublicAttachmentTestCase : public ::testing::Test {};

TEST_F(PublicAttachmentTestCase, TestAttachmentCreationDifferentFile) {
    Attachment a("newfile.pdf");
    EXPECT_EQ(a.filename(), "newfile.pdf");
    EXPECT_FALSE(a.filename().empty());
}

TEST_F(PublicAttachmentTestCase, TestAttachmentReprDifferentFile) {
    Attachment a("readme.md");
    std::string repr = a.repr();
    EXPECT_NE(repr.find("Attachment"), std::string::npos);
}

File: RepoTransBench/target_projects/Python/Rust/fengsp_sender/public_tests/public_test_sender_extended.rs
use fengsp_sender::Attachment;

#[test]
fn test_attachment_creation_different_file() {
    let a = Attachment::new("newfile.pdf");
    assert_eq!(a.filename.as_deref(), Some("newfile.pdf"));
    assert!(std::mem::size_of::<Attachment>() > 0); // instance is not None
}

#[test]
fn test_attachment_repr_different_file() {
    let a = Attachment::new("readme.md");
    let repr = format!("{:?}", a);
    assert!(repr.contains("Attachment"));
}